CMakeLists.txt

```diff
@@ -77,32 +77,26 @@ endif ()
 # capstone
 #
 if (DISASM_CAPSTONE)
+  configure_file(cmake/capstone.cmake.in capstone-download/CMakeLists.txt)
+  execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
+      WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/capstone-download"
+  )
+  execute_process(COMMAND "${CMAKE_COMMAND}" --build .
+      WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/capstone-download"
+  )
+
   string(TOUPPER ${FUNCHOOK_CPU} FUNCHOOK_CPU_UPPER)
 
-  include(ExternalProject)
-  ExternalProject_Add(capstone_src
-    GIT_REPOSITORY    https://github.com/aquynh/capstone.git
-    GIT_TAG           4.0.2
-    GIT_SHALLOW       TRUE
-    CMAKE_ARGS        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-                      -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE}
-                      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-                      -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
-                      -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}
-                      -DCMAKE_POSITION_INDEPENDENT_CODE=ON
-                      -DCAPSTONE_BUILD_SHARED=OFF
-                      -DCAPSTONE_BUILD_STATIC_RUNTIME=OFF
-                      -DCAPSTONE_BUILD_TESTS=OFF
-                      -DCAPSTONE_BUILD_CSTOOL=OFF
-                      -DCAPSTONE_ARCHITECTURE_DEFAULT=OFF
-                      -DCAPSTONE_${FUNCHOOK_CPU_UPPER}_SUPPORT=ON
-    INSTALL_COMMAND   ${CMAKE_COMMAND} --build . --target install
-  )
-  file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/include)
-  add_library(capstone STATIC IMPORTED)
-  set_property(TARGET capstone PROPERTY IMPORTED_LOCATION ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}capstone${CMAKE_STATIC_LIBRARY_SUFFIX})
-  set_property(TARGET capstone PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_BINARY_DIR}/include)
-  list(APPEND FUNCHOOK_DEPS capstone)
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Could you remove this line and add the following code at line 75 and use `add_subdirectory_pic` instead of `add_subdirectory` to compile capstone?

```cmake

function(add_subdirectory_pic source_dir binary_dir)
  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
  add_subdirectory(${source_dir} ${binary_dir})
endfunction(add_subdirectory_pic)

```
I want to apply `CMAKE_POSITION_INDEPENDENT_CODE` only to capstone.

Reply:
Ah sure. This should be done.